src/python/gudhi/delaunay_complex.pyx

```diff
@@ -0,0 +1,249 @@
+    * `'cech'`               - The filtration value of each simplex is computed as a :class:`~gudhi.DelaunayCechComplex`
+    """
+
+    cdef Delaunay_complex_interface * this_ptr
+
+    # Fake constructor that does nothing but documenting the constructor
+    def __init__(self, points=[], weights=None, precision='safe'):
+        """DelaunayComplex constructor.
+
+        :param points: A list of points in d-Dimension.
+        :type points: Iterable[Iterable[float]]
+
+        :param weights: A list of weights. If set, the number of weights must correspond to the number of points.
+        :type weights: Iterable[float]
+
+        :param precision: Delaunay complex precision can be 'fast', 'safe' or 'exact'. Default is 'safe'.
+        :type precision: string
+
+        :raises ValueError: In case of inconsistency between the number of points and weights.
+        """
+
+    # The real cython constructor
+    def __cinit__(self, points = [], weights=None, precision = 'safe'):
+        assert precision in ['fast', 'safe', 'exact'], "Delaunay complex precision can only be 'fast', 'safe' or 'exact'"
+        cdef bool fast = precision == 'fast'
+        cdef bool exact = precision == 'exact'
+
+        # weights are set but is inconsistent with the number of points
+        if weights is not None and len(weights) != len(points):
+            raise ValueError("Inconsistency between the number of points and weights")
+
+        # need to copy the points to use them without the gil
+        cdef vector[vector[double]] pts
+        cdef vector[double] wgts
+        pts = points
+        if weights is not None:
+            wgts = weights
+        with nogil:
+            self.this_ptr = new Delaunay_complex_interface(pts, wgts, fast, exact)
+
+    def __dealloc__(self):
+        if self.this_ptr != NULL:
+            del self.this_ptr
+
+    def _is_defined(self):
+        """Returns true if DelaunayComplex pointer is not NULL.
+         """
+        return self.this_ptr != NULL
+
+    def create_simplex_tree(self, max_alpha_square = float('inf'), filtration = None):
+        """
+        :param max_alpha_square: The maximum alpha square threshold the simplices shall not exceed. Default is set to
+            infinity, and there is very little point using anything else since it does not save time.
+        :type max_alpha_square: float
+        :param filtration: Set this value to `None` (default value) if filtration values are not needed to be computed
+            (will be set to `NaN`). Set it to `alpha` to compute the filtration values with the Alpha complex, or to
+            `cech` to compute the Delaunay Cech complex.
+        :type filtration: string or None
+        :returns: A simplex tree created from the Delaunay Triangulation. The vertices may be not contiguous as some
+            points may be discarded in the triangulation (duplicate points, weighted hidden point, ...). The vertex
+            `k` corresponds to the k-th input point.
```

Comment:
```suggestion
        :returns: A simplex tree created from the Delaunay Triangulation. Some
            points may be discarded in the triangulation (duplicate points, weighted hidden point, ...). The vertex
            `k` corresponds to the k-th input point.
```
(maybe even swap the last 2 sentences)
or if we want to keep "contiguous" without being too confusing
```suggestion
        :returns: A simplex tree created from the Delaunay Triangulation. The vertex `k` corresponds to the k-th input point.
            The vertices may not be numbered contiguously as some
            points may be discarded in the triangulation (duplicate points, weighted hidden point, ...).
```
?

Reply:
I took the second one on 750c0fc7f